src/includes/traits/Plugin/UpdateUtils.php

```diff
@@ -116,9 +116,16 @@ protected function maybeCheckLatestProPackage()
         $product_api_response = wp_remote_post($product_api_url, ['body' => $product_api_input_vars]);
         $product_api_response = json_decode(wp_remote_retrieve_body($product_api_response));
 
+        $this->dismissMainNotice('pro_update_error'); // Clear any previous error notice in case a previous issue has been fixed.
+
+        if (!empty($product_api_response->error)) {
```

Comment:
If the connection fails then `->error` will not be set. That particular property comes from the cometcache.com server, so it the connection fails, `$product_api_response` will be empty.